app/src/App.js

```diff
@@ -51,22 +51,34 @@ const App = () => {
     getUserInfo("timedown");
   }, [googleAccount]);
 
+  if (isAuthenticated) {
+    return <UserDashboard {...{ isAuthenticated, gcal, timedownAccount }} />;
+  }
+
+  return <FooBar {...{ isAuthenticated, gcal }} />;
+};
+
+function UserDashboard({ isAuthenticated, gcal, timedownAccount }) {
   return (
     <main className="App">
-      {isAuthenticated ? (
-        <>
-          <div id="login">
-            <Login {...{ isAuthenticated, gcal }} />
-          </div>
-          <Planner {...{ isAuthenticated, gcal, timedownAccount }} />
-        </>
-      ) : (
+      <>
         <div id="login">
           <Login {...{ isAuthenticated, gcal }} />
         </div>
-      )}
+        <Planner {...{ isAuthenticated, gcal, timedownAccount }} />
+      </>
     </main>
   );
-};
+}
+
+function FooBar({ isAuthenticated, gcal }) {
```

Comment:
What does the fooBar function do?

Reply:
Temp thing I was testing out with @richiethomas, at the time we couldn’t think of a name. He was suggesting I separate the logging-in and logging-out components. We used “Foobar” so I wouldn’t have to go through the time to change the already existing Login component.

[context1](https://github.com/sunnybrie/Timedown/pull/9#discussion_r630683877)
[context2](https://github.com/sunnybrie/Timedown/pull/9#discussion_r630685986)

Reply:
Actually I think there's a way to make this simpler than we did on the first pass.  Right now we've duplicated the wrapper divs like `main className="App"`.  As an alternative, a single return statement instead of the guard clauses could work like this:

```
  return (
    <main className="App">
      <>
        <div id="login">
          <Login {...{ isAuthenticated, gcal }} />
        </div>
        { isAuthenticated && <Planner {...{ isAuthenticated, gcal, timedownAccount }} />}
      </>
    </main>
  );
```

This would also remove the need for `UserDashboard` and `Foobar`. :-)

I do still think the `Logout` functionality should be extracted out of the `Login` component and into its own component for the sake of clarity.  But if `Login` is going to be returned either way then we *could* do the above instead. :-)